api/src/main/java/org/openmrs/api/db/hibernate/HibernatePersonDAO.java

```diff
@@ -72,6 +73,121 @@ public void setSessionFactory(SessionFactory sessionFactory) {
 		this.sessionFactory = sessionFactory;
 	}
 	
+	/**
+	 * This method executes a Lucene search on persons based on the soundex filter with one search name given
+	 * 
+	 * @param name a person should match using soundex representation
+	 * @param birthyear the birthyear the searched person should have 
+	 * @param includeVoided true if voided person should be included 
+	 * @param gender of the person to search for 
+	 * @return the set of Persons that match the search criteria 
+	 */
+	private Set<Person> executeSoundexOnePersonNameQuery(String name, Integer birthyear, boolean includeVoided , String gender) {
+		// Use the Soundex Search Mode for each query
+		String matchMode = Context.getAdministrationService().getGlobalProperty(OpenmrsConstants.GLOBAL_PROPERTY_PATIENT_SEARCH_MATCH_MODE);
+		Context.getAdministrationService().setGlobalProperty(OpenmrsConstants.GLOBAL_PROPERTY_PATIENT_SEARCH_MATCH_MODE, OpenmrsConstants.GLOBAL_PROPERTY_PATIENT_SEARCH_MATCH_SOUNDEX);
+		
+		
+		PersonLuceneQuery personLuceneQuery = new PersonLuceneQuery(sessionFactory);
+		String query = LuceneQuery.escapeQuery(name);
+		int maxResults = HibernatePersonDAO.getMaximumSearchResults();
+		LinkedHashSet<Person> people = new LinkedHashSet<>();
+		
+		LuceneQuery<PersonName> luceneQuery = personLuceneQuery.getSoundexPersonNameQuery(query, birthyear, false, gender);;
+		ListPart<Object[]> names = luceneQuery.listPartProjection(0, maxResults, "person.personId");
+		names.getList().forEach(x -> people.add(getPerson((Integer) x[0])));
+		
+		Context.getAdministrationService().setGlobalProperty(OpenmrsConstants.GLOBAL_PROPERTY_PATIENT_SEARCH_MATCH_MODE, matchMode);
+		
+		return people; 
+	}
+	
+	/**
+	 * This method executes a Lucene search on persons based on the soundex filter with two search names given
+	 * 
+	 * @param name1 firstName to be searched
+	 * @param name2 last name to be searched 
+	 * @param birthyear the birthyear the searched person should have 
+	 * @param includeVoided true if voided person should be included 
+	 * @param gender of the person to search for 
+	 * @return the set of Persons that match the search criteria
+	 */
+	private Set<Person> executeSoundexTwoPersonNameQuery(String name1, String name2,  Integer birthyear, boolean includeVoided , String gender) {
+		// Use the Soundex Search Mode for each query
+		String matchMode = Context.getAdministrationService().getGlobalProperty(OpenmrsConstants.GLOBAL_PROPERTY_PATIENT_SEARCH_MATCH_MODE);
+		Context.getAdministrationService().setGlobalProperty(OpenmrsConstants.GLOBAL_PROPERTY_PATIENT_SEARCH_MATCH_MODE, OpenmrsConstants.GLOBAL_PROPERTY_PATIENT_SEARCH_MATCH_SOUNDEX);
```

Comment:
What happens when two people search for patients at the same time in different browsers?

Reply:
Honestly, I don't know. Of all the options it seems like this was the smoothest way. 
Perhaps there is a constant that is bound to a session? 
Or there is a near way to set the choosen Lucene Analyzer that I was not aware of

Reply:
I asked in talk if we can get some support deciding what the approach would be. Hope you do not mind!

Reply:
Not at all!